### PR TITLE
Add test suite

### PR DESCRIPTION
**Pull request by allevo**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "generate-function": "^2.0.0"
   },
   "devDependencies": {
+    "pre-commit": "^1.2.2",
```
Comment (mafintosh) [inline package.json]: what does this do? :)

Comment (allevo) [reply package.json]: Add a git hook in order to run `npm test` before commit. That force the developer to create "non-broken" commits https://www.npmjs.com/package/pre-commit

Anyway I'm not forcing anyone. If you want I'll remove it

Comment (mafintosh) [reply package.json]: Ya I'd prefer not to include this for now.

Comment (allevo) [reply package.json]: Commit removed

**Pull request merged**